### PR TITLE
fix: no attribute 'summary_polyline'

### PR DESCRIPTION
**Pull request by shuxiao9058**
The healthfit synced Apple Watch record to Strava but no `summary_polyline`.

```
Run python scripts/strava_sync.py *** *** ***
Access ok
Start syncing
something wrong with 7796808006
'NoneType' object has no attribute 'summary_polyline'
.something wrong with 7804774185
'NoneType' object has no attribute 'summary_polyline'
.something wrong with 7817409219
'NoneType' object has no attribute 'summary_polyline'
.
```

### CODE REVIEW

**Review by shaonianche — APPROVED** [review]

**Comment by shaonianche** [comment]
LGTM!

**Comment by shaonianche** [comment]
> The healthfit synced Apple Watch record to Strava but no summary_polyline.

I had the same problem with my healthfit sync. Lost map data after apple watch 8.6

File: scripts/generator/db.py [hunk 1]
```diff
@@ -109,7 +109,9 @@ def update_or_create_activity(session, run_activity):
                 location_country=location_country,
                 average_heartrate=run_activity.average_heartrate,
                 average_speed=float(run_activity.average_speed),
-                summary_polyline=run_activity.map.summary_polyline,
+                summary_polyline=(
+                    run_activity.map and run_activity.map.summary_polyline or ""
```
Comment (yihong0618) [inline scripts/generator/db.py:113]: what is the `and` here?

Comment (shuxiao9058) [reply scripts/generator/db.py:113]: if has `map` member which is a valid object, then  get return`summary_polyline `, otherwise get empty string.

Comment (yihong0618) [reply scripts/generator/db.py:113]: IMO
the logic is for the `and` is that
if run_activity.map.summary_polyline is not None or not empty string 
summary_polyline = run_activity.map.summary_polyline
and ignore  run_activity.map am I right?

Comment (yihong0618) [reply scripts/generator/db.py:113]: make sence.

Comment (yihong0618) [reply scripts/generator/db.py:113]: summary_polyline = run_activity.map.summary_polyline if run_activity.map else ""
maybe more easy to understand?

Comment (shuxiao9058) [reply scripts/generator/db.py:113]: if `run_activity.map` is always not None then can remove run_activity.map validate.

**Review by yihong0618 — APPROVED** [review]

File: scripts/generator/db.py (hunk 1 above)
Comment (yihong0618) [reply scripts/generator/db.py:113]: @shuxiao9058 
This code is good for me.
I wonder if there is an easier way to understand.

if not will merge it now.

Comment (yihong0618) [reply scripts/generator/db.py:113]: merging it now.

**Comment by yihong0618** [comment]
Thanks

**Comment by yihong0618** [comment]
@shuxiao9058 just invited you as a collaborator.
A lot of thanks for your work as usual

**Pull request merged**